core/gotemplate/engine.go

```diff
@@ -64,14 +64,14 @@ func (e *engine) Render(ctx context.Context, name string, data interface{}) (io.
 	ctx, span := trace.StartSpan(ctx, "gotemplate/Render")
 	defer span.End()
 
-	lock.Lock()
 	if e.debug || e.templates == nil {
+		lock.Lock()
```

Comment:
This will lead to a race condition when e.templates is nil, eventually triggering e.loadTemplates multiple times. Move lock.Lock() up, and call lock.Unlock() both in error and non-error case.